gdx/src/com/badlogic/gdx/utils/IdentityMap.java

```diff
@@ -500,6 +500,38 @@ private int hash3 (int h) {
 		return (h ^ h >>> hashShift) & mask;
 	}
 
+	@Override
+	public int hashCode () {
+		int h = 0;
+		for (Entry<K, V> entry : entries()) {
+			h += (entry.key != null ? entry.key.hashCode() : 0) * 31;
+			h += (entry.value != null ? entry.value.hashCode() : 0);
+		}
+		return h;
+	}
+
+	@Override
+	public boolean equals (Object obj) {
+		if (obj == this) return true;
+		if (!(obj instanceof IdentityMap)) return false;
+		IdentityMap<K, V> other = (IdentityMap) obj;
+		if (other.size != size) return false;
+		for (Entry<K, V> entry : entries()) {
+			V otherValue = other.get(entry.key);
+			if (entry.value == null) {
+				if (!other.containsKey(entry.key) || other.get(entry.key) != null) {

```

Comment:
I got rid of `otherValue`, since it was unused. It seems like the "unused" warnings have been suppressed. The case where `entry.value == null` is pretty rare, so I guess there is no point in optimizing it further. If we want this to run as fast as possible, then the backing arrays should probably be used instead of `Entries`.